lib/src/main/kotlin/com/lemonappdev/konsist/core/declaration/KoDoc.kt

```diff
@@ -0,0 +1,178 @@
+package com.lemonappdev.konsist.core.declaration
+
+import org.jetbrains.kotlin.kdoc.psi.api.KDocElement
+import org.jetbrains.kotlin.kdoc.psi.impl.KDocSection
+
+class KoDoc(private val kDocElement: KDocElement) {
+    val text: String by lazy { kDocElement.text }
+
+    val description by lazy {
+        kDocElement
+            .children
+            .filterIsInstance<KDocSection>()
+            .first()
+            .text
+            .substringBefore("@")
+            .split("\n")
+            .map {
+                it
+                    .substringAfter("*")
+                    .removePrefix(" ")
+            }
+            .toMutableList()
+            .also { it.removeIf { element -> element.isBlank() } }
+            .joinToString("\n")
+    }
+
+    private val tags by lazy {
+        val firsts = kDocElement
+            .children
+            .filterIsInstance<KDocSection>()
+            .first()
+            .text
+            .replaceBefore('@', "")
+            .split("\n")
+            .map {
+                it
+                    .substringAfter("*")
+                    .removePrefix(" ")
+            }
+            .toMutableList()
+            .also { it.removeIf { element -> element.isBlank() } }
+
+        val others = kDocElement
+            .children
+            .filterIsInstance<KDocSection>()
+            .toMutableList()
+            .also { it.removeFirst() }
+            .map { it.text }
+            .flatMap {
+                it.split("\n")
+                    .map { line ->
+                        line
+                            .substringAfter("*")
+                            .removePrefix(" ")
+                    }
+                    .toMutableList()
+                    .also { line -> line.removeIf { element -> element.isBlank() } }
+            }
+
+        firsts + others
+    }
```

Comment:
W `description` i dwa razy w `tags` parsowanie jest brzydkie i trochę podobne, ale jednak inne. I nie wiem jak to jakoś uprościć